### PR TITLE
Improve error message when uploading unsupported datatypes

### PR DESCRIPTION
**Pull request by khoaguin**
Issue: https://github.com/OpenMined/Heartbeat/issues/1662

Now:
image

### CODE REVIEW

File: packages/syft/src/syft/client/datasite_client.py [hunk 1]
```diff
@@ -146,6 +147,8 @@ def upload_dataset(self, dataset: CreateDataset) -> SyftSuccess | SyftError:
                     res = twin._save_to_blob_storage(allow_empty=contains_empty)
                     if isinstance(res, SyftError):
                         return res
+                except SyftException as se:
+                    return SyftError(message=f"{se}")
```
Comment (shubham3121) [inline packages/syft/src/syft/client/datasite_client.py:151]: Is this line required ? Doesn't Exception catch everything ?

File: packages/syft/src/syft/service/action/action_object.py [hunk 2]
```diff
@@ -1412,6 +1414,25 @@ def from_obj(
...
+                f"Error when creating action object for {syft_action_data}.\n"
+                f"Unsupported data type: '{obj_type.__module__}.{obj_type.__name__}'"
+            )
+        can_be_serialized = SyftObjectRegistry.has_serde_class(canonical_name, version)
+        if not can_be_serialized:
+            warning = SyftWarning(
+                message=f"Object of type '{obj_type.__module__}.{obj_type.__name__}' "
+                f"is not supported by Syft serialization."
+            )
+            display(warning)
```
Comment (shubham3121) [inline packages/syft/src/syft/service/action/action_object.py]: Should we throw an Exception ?

File: packages/syft/src/syft/types/syft_object_registry.py [hunk 3]
```diff
@@ -76,7 +76,9 @@ def get_canonical_name_version(cls, obj: Any) -> tuple[str, int]:
         obj_type = type(obj)
         if obj_type in cls.__type_to_canonical_name__:
             return cls.__type_to_canonical_name__[obj_type]
-        raise ValueError(f"Could not find canonical name for {obj}")
+        raise ValueError(
+            f"Could not find canonical name for '{obj_type.__module__}.{obj_type.__name__}'"
```
Comment (shubham3121) [inline packages/syft/src/syft/types/syft_object_registry.py:80]: 👍🏾

File: packages/syft/src/syft/service/action/action_object.py (hunk 2 above)
Comment (khoaguin) [inline packages/syft/src/syft/service/action/action_object.py]: We can remove this

**Review by shubham3121 — APPROVED** [review]
Just a minor comment, otherwise looks good.
Great work @khoaguin !! 🎉 💯

**Pull request merged**